quickjs.c

```diff
@@ -1694,7 +1694,7 @@ static void *js_def_malloc(JSMallocState *s, size_t size)
     /* Do not allocate zero bytes: behavior is platform dependent */
     assert(size != 0);
 
-    if (unlikely(s->malloc_size + size > s->malloc_limit))
+    if (unlikely(s->malloc_limit > 0 && s->malloc_size + size > s->malloc_limit))
         return NULL;
```

Comment:
This is inefficient: 2 tests instead of 1.  You could use this instead:
```
if (unlikely(s->malloc_size + size >= s->malloc_limit - 1))
```
The idea is that `s->malloc_limit - 1` will be `SIZE_MAX` if `s->malloc_limit == 0`.  An explanatory comment would be helpful.